BuilderPattern.java

```diff
@@ -0,0 +1,54 @@
+package Ex4;
+public class BuilderPattern{
+	public static void main(String[] args) {
+        Car myCar=new Car.BuildCar().setCompany("Honda").setModel("Civic").setEngine("Petrol").setTire("MRF").build();
+        System.out.println(myCar);
+	}
+}
+class Car{
+    class CarBuilder{
+    String tire, engine, model, company;
+    public CarBuilder(String company,String model,String engine,String tire) {
+    this.tire=tire;
+    this.engine=engine;
+    this.model=model;
+    this.company=company;
+ }
+ 
+    public String toString() {
+        return "CarBuilder[Company="+company+",Model="+model+",Engine="+engine+",Tire="+tire+"]";
+    }
+ } 
+	CarBuilder carbuilder;
+	public Car(BuildCar buildCar) {
+        carbuilder=new CarBuilder(buildCar.company,buildCar.model,buildCar.engine,buildCar.tire);
+	}
+	
+	public String toString() {
+        return "Car[carbuilder="+carbuilder+"]";
+	}
+	
+	
+	static class BuildCar{
```

Comment:
whether static is needed here?

Reply:
Yes Na...because I am using an inner class of class Car...if we make the class as static we can directly access the methods of the inner classes without creating instance of the class Car.